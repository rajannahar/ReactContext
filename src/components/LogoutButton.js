import React, {useContext} from 'react'
import { AuthContext } from '../AuthContext/AuthContext'

const LogoutButton = () => {
  const { logout } = useContext(AuthContext)
  return(
    <button data-testid="logoutButton"  onClick={() => logout()}>Logout</button>
  )
}

export default LogoutButton;