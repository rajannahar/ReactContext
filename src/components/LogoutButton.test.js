import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react'
import { getByTestId } from '@testing-library/dom'
import LogoutButton from './LogoutButton';

test('should render logout button', () => {
  const { getByTestId } = render(<LogoutButton />)
  expect(getByTestId('logoutButton')).toBeTruthy()
})

test('should have text "Logout"', () => {
  const { getByTestId } = render(<LogoutButton />)
  const button = getByTestId('logoutButton')
  expect(button.innerHTML).toEqual('Logout')
})
