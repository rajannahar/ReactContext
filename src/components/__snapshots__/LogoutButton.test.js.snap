// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    data-testid="logoutButton"
  >
    Logout
  </button>
</DocumentFragment>
`;
